src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -150,38 +133,38 @@ public void postPatronAccountItemRenewByIdAndItemId(String id, String itemId,
   public void postPatronAccountItemHoldByIdAndItemId(String id, String itemId,
       Hold entity, Map<String, String> okapiHeaders,
       Handler<AsyncResult<javax.ws.rs.core.Response>> asyncResultHandler, Context vertxContext) {
-    final JsonObject holdJSON = new JsonObject()
-        .put(JSON_FIELD_ITEM_ID, itemId)
-        .put("requesterId", id)
-        .put("requestType", "Hold")
-        .put(JSON_FIELD_REQUEST_DATE, new DateTime(entity.getRequestDate(), DateTimeZone.UTC).toString())
-        .put("fulfilmentPreference", JSON_VALUE_HOLD_SHELF)
-        .put(JSON_FIELD_PICKUP_SERVICE_POINT_ID, entity.getPickupLocationId());
 
-    if (entity.getExpirationDate() != null) {
-      holdJSON.put(JSON_FIELD_REQUEST_EXPIRATION_DATE,
-          new DateTime(entity.getExpirationDate(), DateTimeZone.UTC).toString());
-    }
+    final HttpClientInterface httpClient = LookupsUtils.getHttpClient(okapiHeaders);
+    RequestObjectFactory requestFactory = new RequestObjectFactory(okapiHeaders);
 
-    final HttpClientInterface httpClient = getHttpClient(okapiHeaders);
-    try {
-      httpClient.request(HttpMethod.POST, Buffer.buffer(holdJSON.toString()), "/circulation/requests", okapiHeaders)
-          .thenApply(this::verifyAndExtractBody)
-          .thenAccept(body -> {
-            final Item item = getItem(itemId, body.getJsonObject(JSON_FIELD_ITEM));
-            final Hold hold = getHold(body, item);
-            asyncResultHandler.handle(Future.succeededFuture(PostPatronAccountItemHoldByIdAndItemIdResponse.respond201WithApplicationJson(hold)));
-            httpClient.closeClient();
-          })
-          .exceptionally(throwable -> {
-            asyncResultHandler.handle(handleItemHoldPOSTError(throwable));
+    requestFactory.createRequestByItem(id, itemId, entity)
+      .thenCompose(holdJSON -> {
+        try {
+          if (holdJSON == null) {
+            asyncResultHandler.handle(handleItemHoldPOSTError(new Exception("Cannot find a valid request type for this item")));
```

Comment:
I wonder if a 500 error is the right code for this. What does circulation return when we try to create a request on an item but none of the request types apply?

Reply:
Good point. mod-circulation returns a 422 validation error

Reply:
It's returning a 422 now.